### ISSUE TITLE
"PHP files in the Backdrop files directory can be executed." test incorrect

### ISSUE THREAD

**Issue by yorkshire-pudding** [backdrop-contrib/security_review#1]
**Steps to reproduce**
1. Run Security Review check
2. View Details for "PHP files in the Backdrop files directory can be executed." 

**Expected result**
Should pass as this has never been changed and has permissions 444
Also running `find . -type f -executable` in the files directory returned no results

**Actual result**

![image](https://user-images.githubusercontent.com/20357797/149304687-d095a6fa-490c-4df0-ae71-1ee5a693e145.png)
fails with what seems to be an incorrect failure.  

Details says:
>Executable PHP in files directory
>The Backdrop files directory is for user-uploaded files and by default provides some protection against a malicious user executing arbitrary PHP code against your site.
>
>Read more about the risk of PHP code execution on Drupal.org.
>
>The .htaccess file exists but does not contain the correct content. It is possible it's been maliciously altered.

As the first line says Turn off all options we don't need, then comparing this line does not seem appropriate

### Actual .htaccess file
```
# Turn off all options we don't need.
Options None
Options +SymLinksIfOwnerMatch

# Set the catch-all handler to prevent scripts from being executed.
SetHandler Drupal_Security_Do_Not_Remove_See_SA_2006_006
<Files *>
  # Override the handler again if we're run later in the evaluation list.
  SetHandler Drupal_Security_Do_Not_Remove_See_SA_2013_003
</Files>

# If we know how to do it safely, disable the PHP engine entirely.
<IfModule mod_php5.c>
  php_flag engine off
</IfModule>
```

### function from /core/includes/file.inc
```
/**
 * Returns the standard .htaccess lines that Backdrop adds to file directories.
 *
 * @param $private
 *   (Optional) Set to FALSE to return the .htaccess lines for an open and
 *   public directory. The default is TRUE, which returns the .htaccess lines
 *   for a private and protected directory.
 *
 * @return
 *   A string representing the desired contents of the .htaccess file.
 *
 * @see file_save_htaccess()
 */
function file_htaccess_lines($private = TRUE) {
  $lines = <<<EOF
# Turn off all options we don't need.
Options -Indexes -ExecCGI -Includes

# Set the catch-all handler to prevent scripts from being executed.
SetHandler Drupal_Security_Do_Not_Remove_See_SA_2006_006
<Files *>
  # Override the handler again if we're run later in the evaluation list.
  SetHandler Drupal_Security_Do_Not_Remove_See_SA_2013_003
</Files>

# If we know how to do it safely, disable the PHP engine entirely.
<IfModule mod_php5.c>
  php_flag engine off
</IfModule>
<IfModule mod_php7.c>
  php_flag engine off
</IfModule>
EOF;

  if ($private) {
    $lines .= <<<EOF

# Deny all requests from Apache 2.4+.
<IfModule mod_authz_core.c>
  Require all denied
</IfModule>

# Deny all requests from Apache 2.0-2.2.
<IfModule !mod_authz_core.c>
  Deny from all
</IfModule>

EOF;
  }

  return $lines;
}
```

**Comment by oadaeh** [comment]
@yorkshire-pudding it looks as though your .htaccess file is missing this part from the code:
```
<IfModule mod_php7.c>
  php_flag engine off
</IfModule>
```
If you add that to the end of your .htaccess file, that test will pass. It did for me.

**Comment by oadaeh** [comment]
BTW, that part was added 8 months ago: https://github.com/backdrop/backdrop/commit/492c6b68478cbb8f8e199dd82e815b5e04363792

**Comment by yorkshire-pudding** [comment]
@oadaeh thanks.
I've copied the whole lot from file.inc replacing what was in there and I still get the test failure. Can't understand this as 

```
# Turn off all options we don't need.
Options -Indexes -ExecCGI -Includes

# Set the catch-all handler to prevent scripts from being executed.
SetHandler Drupal_Security_Do_Not_Remove_See_SA_2006_006
<Files *>
  # Override the handler again if we're run later in the evaluation list.
  SetHandler Drupal_Security_Do_Not_Remove_See_SA_2013_003
</Files>

# If we know how to do it safely, disable the PHP engine entirely.
<IfModule mod_php5.c>
  php_flag engine off
</IfModule>
<IfModule mod_php7.c>
  php_flag engine off
</IfModule>
```

Given the test code is

```
    $contents = file_get_contents($directory . '/.htaccess');
    // Text from includes/file.inc.
    $expected = file_htaccess_lines(FALSE);
    if (trim($contents) !== trim($expected)) {
      $result = FALSE;
      $check_result_value[] = 'incorrect_htaccess';
    }
```
I'm at a loss to understand how this test is still failing. 

There is another issue here.  The file should not be writeable for security so how does it get updated if backdrop updates?  For sitebuilders to have a site that passes one day then doesn't could be quite confusing.

**Comment by oadaeh** [comment]
I assume you've checked this already, but I feel compelled to ask, is it pointing to your public or private .htaccess file? Maybe it's referring to your private one and you're fixing the public one? Other than that, I don't know without going down the rabbit hole myself. I do know that making that change on my sites did fix the notice, so I assume there's _something_ going on on your system that's triggering the notice for you.

> There is another issue here. The file should not be writeable for security so how does it get updated if backdrop updates? For sitebuilders to have a site that passes one day then doesn't could be quite confusing.

They don't get updated. Mine didn't. I had to manually change them.
There is this: https://www.drupal.org/project/htaccess
But, A) it seems to only be for the main .htaccess file (though I suppose it could be modified to add the files ones), and B) I don't think I would leave that task in the hands of a site builder anyway.

**Comment by yorkshire-pudding** [comment]
Hi @oadaeh - Yes, it's just public files - I haven't implemented private files on this site as there is no need.

Thanks anyway - I'll just ignore it for now, but leave this ticket open in case anyone else has any ideas why this is

**Comment by yorkshire-pudding** [comment]
@oadaeh - any chance, please, that you could review the PR in core that will fix this? - https://github.com/backdrop/backdrop/pull/3913

**Comment by yorkshire-pudding** [comment]
As this has been merged and included in release 1.21.2 (see https://github.com/backdrop/backdrop-issues/issues/5513).  Security Review now passes on this test, so closing this issue.
